src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestAttributeValueResolver.php

```diff
@@ -35,6 +35,14 @@ public function supports(Request $request, ArgumentMetadata $argument)
      */
     public function resolve(Request $request, ArgumentMetadata $argument)
     {
-        yield $request->attributes->get($argument->getName());
+        if ('float' === $argument->getType()) {
```

Comment:
switch instead to prevent calling getType n-times?

Reply:
good idea, I started with that and wondered if it was clearer or not (must add `break`, since no early `return` is possible in generators).

Reply:
fixed